lib/src/contextual-menu/GroupItem.tsx

```diff
@@ -1,32 +1,36 @@
 import React, { useContext, useMemo, useState } from "react";
 import styled from "styled-components";
 import CoreTokens from "../common/coreTokens";
-import { GroupItemProps } from "./types";
+import { GroupItemProps, ItemWithId } from "./types";
 import MenuItem from "./MenuItem";
 import ItemAction from "./ItemAction";
 import { ContextualMenuContext } from "./ContextualMenu";
 import DxcIcon from "../icon/Icon";
 
 const isGroupSelected = (items: GroupItemProps["items"], selectedItemId: number): boolean =>
-  items.some((item) => ("id" in item ? item.id === selectedItemId : isGroupSelected(item.items, selectedItemId)));
+  items.some((item) => {
+    if ("items" in item) return isGroupSelected(item.items, selectedItemId);
+    else if (selectedItemId !== -1) return item.id === selectedItemId;
+    else return (item as ItemWithId).selectedByDefault ? true : false;
```

Comment:
This ternary operator is redundant (selectedByDefault is already a boolean).

Reply:
`selectedByDefault` can be `undefined` (optional prop) and our function always returns something. We need to ensure that even if it is undefined, it returns a boolean value (in this case, `false`).

Reply:
`undefined` will be treated as falsy inside `some` though.

Anyway, in these kind of scenarios where you want to cast something to boolean, what is considered the best practice is using double `!`.
```
else return !!(item as ItemWithId).selectedByDefault
```

Reply:
I didn't know that some treated undefined as false! I removed it.